File: src/main/java/com/github/yuppieflu/stats/system/StatsConfiguration.java
package com.github.yuppieflu.stats.system;

import com.github.yuppieflu.stats.service.StorageService;
import com.github.yuppieflu.stats.service.bucket.BucketsStorageService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.time.Clock;

@Configuration
public class StatsConfiguration {

    @Bean
    public StorageService storageService() {
        return new BucketsStorageService(Clock.systemUTC());
    }
}

File: src/main/kotlin/com/github/yuppieflu/stats/system/StatsConfiguration.kt
package com.github.yuppieflu.stats.system

import com.github.yuppieflu.stats.service.StorageService
import com.github.yuppieflu.stats.service.bucket.BucketsStorageService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

import java.time.Clock

@Configuration
open class StatsConfiguration {

    @Bean
    open fun storageService(): StorageService {
        return BucketsStorageService(Clock.systemUTC())
    }
}
